File: trackup/Classes/TrackupDocument.m
//
//  TrackupDocument.m
//  trackup
//
//  Created by Vincent Tourraine on 01/02/15.
//  Copyright (c) 2015 Studio AMANgA. All rights reserved.
//

#import "TrackupDocument.h"

@implementation TrackupDocument

@end


@implementation TrackupVersion

- (BOOL)isInProgress {
    for (TrackupItem *item in self.items) {
        if (item.state != TrackupItemStateUnknown) {
            return YES;
        }
    }

    return NO;
}

@end


@implementation TrackupItem

@end

File: TrackupCore/Sources/TrackupCore/TrackupDocument.swift
//
//  TrackupDocument.swift
//  trackup
//
//  Created by Vincent Tourraine on 29/11/16.
//  Copyright © 2016-2022 Studio AMANgA. All rights reserved.
//

import Foundation

struct K {
    static let roadmapTitle = "Roadmap"
}

public struct TrackupDocument: Codable {
    public var title: String = ""
    public var versions: [TrackupVersion] = []
    public var website: URL?
}

public struct TrackupVersion: Codable {
    public var title: String = ""
    public var items: [TrackupItem] = []
    public var createdDate: DateComponents?

    func inProgress() -> Bool {
        for item in self.items {
            if (item.state != .unknown) {
                return true
            }
        }

        return false
    }
}

public struct TrackupItem: Codable {
    public var title: String = ""
    public var state: TrackupItemState = .unknown
    public var status: TrackupItemStatus = .unknown
}

public enum TrackupItemState: String, Codable {
    case unknown
    case todo
    case done
}

public enum TrackupItemStatus: String, Codable {
    case unknown
    case major
}
